crates/rpc-replay/src/transactions.rs

```diff
@@ -90,21 +98,196 @@ pub fn starknet_rs_to_blockifier(
+                            fee_data_availability_mode: da_mode_core_to_api(tx.fee_data_availability_mode),
+                            paymaster_data: starknet_api::transaction::PaymasterData(tx.paymaster_data.clone()),
+                            account_deployment_data: starknet_api::transaction::AccountDeploymentData(
+                                tx.account_deployment_data.clone(),
+                            ),
+                        },
+                    );
+
+                    (tx_hash, api_tx, tx.class_hash)
+                }
+
+                _ => unimplemented!("DeclareTransaction V0 not supported"),
+            };
+
+            // TODO: improve this to avoid retrieving this twice. Already done in main.rs from prove_block
+            let starknet_contract_class = provider.get_class(BlockId::Number(block_number), class_hash).await?;
+            let generic_sierra_cc = match starknet_contract_class {
+                starknet::core::types::ContractClass::Sierra(flattened_sierra_cc) => {
+                    GenericSierraContractClass::from(flattened_sierra_cc)
+                }
+                starknet::core::types::ContractClass::Legacy(_) => {
+                    unimplemented!("Fixme: Support legacy contract class")
+                }
+            };
+
+            let flattened_sierra = generic_sierra_cc.clone().to_starknet_core_contract_class()?;
+            let contract_class = generic_sierra_cc.compile()?.get_blockifier_contract_class()?.clone();
+
+            let class_info = ClassInfo::new(
+                &contract_class.into(),
+                flattened_sierra.sierra_program.len(),
+                flattened_sierra.abi.len(),
+            )?;
+
+            let declare = blockifier::transaction::transactions::DeclareTransaction::new(api_tx, tx_hash, class_info)?;
+
+            blockifier::transaction::transaction_execution::Transaction::AccountTransaction(
+                AccountTransaction::Declare(declare),
+            )
         }
-        Transaction::L1Handler(_tx) => {
-            unimplemented!("starknet_rs_tx_to_blockifier() with L1Handler txn");
+        Transaction::L1Handler(tx) => {
+            let tx_hash = TransactionHash(tx.transaction_hash);
+            let api_tx = starknet_api::transaction::L1HandlerTransaction {
+                version: starknet_api::transaction::TransactionVersion(tx.version),
+                nonce: starknet_api::core::Nonce(Felt::from(tx.nonce)),
+                contract_address: starknet_api::core::ContractAddress(
+                    PatriciaKey::try_from(tx.contract_address).unwrap(),
+                ),
+                entry_point_selector: starknet_api::core::EntryPointSelector(tx.entry_point_selector),
+                calldata: starknet_api::transaction::Calldata(Arc::new(tx.calldata.clone())),
+            };
+
+            // TODO: Check Fee value
+            let fee = Fee(0);
```

Comment:
Looking at the explorer, this seems to be correct. Replace the TODO by a comment saying that fees are 0 for L1 handler txs.